torch/distributed/_tools/memory_tracker.py

```diff
@@ -176,7 +187,37 @@ def show_traces(self) -> None:
                 plt.plot(
                     [marker, marker], [min_val, max_val], "k-", lw=2, label=marker_name
                 )
-        plt.legend()
+
+    def save_stats(self, path: str) -> None:
+        """
+        Save the stats using pickle during runtime if users want to plot the traces
+        in other places like Bento.
+        """
+        import pickle
+
+        stats = {
+            "memories_allocated": self.memories_allocated,
+            "memories_active": self.memories_active,
+            "memories_reserved": self.memories_reserved,
+            "markers": self._markers,
+        }
+
+        with open(path, "wb") as f:
+            pickle.dump(stats, f)
+
+    def load(self, path: str) -> None:
+        """
+        Load the pickled memory stats to plot the traces or print the summary.
+        """
+        import pickle
+
+        with open(path, "rb") as f:
+            stats = pickle.load(f)
+
```

Comment:
Is there any need to add validation of the loaded object (possibly in follow-ups)?

Reply:
feel validation here is not very helpful, as it will throw errors when access field if the loaded data is not valid